src/carve/main.clj

```diff
@@ -61,11 +62,35 @@
     (validate-opts! opts)
     opts))
 
+(defn- slurp-config []
+  (let [config-file (io/file ".carve/config.edn")]
+    (when (.exists config-file)
+      (edn/read-string (slurp config-file)))))
+
+(defn run!
+  "Programmatic API especially apt for being invoked within an existing, vanilla JVM.
+
+  The defaults are data-oriented and (generally) favor a side-effect-free funcionality.
+
+  If you pass `opts` as map (empty or not), it will be used as the basis for configuration.
+  If you pass `nil` instead, a .carve/config.edn will be read and parsed, if it exists.
+
+  In both cases, the mentioned defaults will be assoc'ed for absent keys."
+  [maybe-opts]
+  (let [{:keys [dry-run interactive silent] :as opts} (or maybe-opts (slurp-config))
+        format-path [:report :format]
+        opts (cond-> opts
+               (nil? dry-run)                            (assoc :dry-run true)
```

Comment:
Any reason you haven't got this in the `or` clause? I think that's safer since when people call this API with `(run! {:dry-run nil})` you will override the value.

Reply:
I think the `nil?` handling was more important in a previous iteration. Now there isn't much difference (user-passed nil values will now fail under the validation call... which is pretty desirable)